### PR TITLE
feat(simulator): 싸움꾼(Brawler/HPTank) trait

### PR DESCRIPTION
**Pull request by keyy1315**
## Summary

TFT17_HPTank (싸움꾼/Brawler) trait 구현 — teamwide +5% maxHp + 싸움꾼 본인 추가 % maxHp.

## Spec

| Tier | minUnits | TeamwideBonus | HealthBonus (싸움꾼) |
|------|----------|--------------|--------------------|
| (2) | 2 | +5% | +20% |
| (4) | 4 | +5% | +40% |
| (6) | 6 | +5% | +60% |

싸움꾼 챔프 (7명): Maokai, Urgot, Gragas, Chogath, TahmKench, RekSai, Pantheon.

## Implementation

- `applyBrawlerEffects` (combatLoop.ts): `TeamwideBonus` 모든 아군 + `HealthBonus` 싸움꾼 본인 multiplicative 가산.
- `maxHp` 가산 시 `currentHp` 도 비례 증가 (전투 시작 시점 풀체).
- 호출 위치: `applyTimebreakerEffects` 다음, `applyVanguardEffects` 전.
- 정령족(Astronaut) flat `BonusHealth` 적용 후 multiplicative — 정령족+싸움꾼 동시 보유 챔프 없어 corner case 영향 없음.

## Test Plan

- [x] `tests/unit/simulator/brawler-trait.test.ts` 추가 (3 cases)
  - (2) tier teamwide +5% + 싸움꾼 +20%: 비-싸움꾼 1.05, 싸움꾼 1.25
  - (4) tier 싸움꾼 +40%: Maokai 1.45
  - Inactive (1명): teamwide 안 됨 → 2명일 때 +5%
- [x] 4-gate (lint/typecheck/test/build) 통과
- [x] 368 unit tests 모두 통과

## Calibration

| Game | 이전 winnerMatch | 현재 | Δ |
|------|-----|-----|---|
| 23일 | 45.5% | 45.5% | 0 |
| 24일 | 76.2% (avgDmgErr -16.9%) | 76.2% (avgDmgErr -16.6%) | dmg err 소폭 개선 |

## Future Work

- 싸움꾼 ability buff 별도 (개별 ability description 의 능력치는 ability 레벨에서 처리).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/lib/simulator/engine/combatLoop.ts [hunk 1]
```diff
@@ -1966,6 +1996,10 @@ export function simulateCombat(
   applySniperEffects(enemyActiveTraits, enemies);
   applyTimebreakerEffects(playerActiveTraits, playerUnits);
   applyTimebreakerEffects(enemyActiveTraits, enemies);
+  // 싸움꾼 +HP — multiplicative. Astronaut flat (+BonusHealth) 적용 후 multiply.
+  // 정령족+싸움꾼 동시 보유 챔프 없어 corner case 영향 없음.
+  applyBrawlerEffects(playerActiveTraits, playerUnits);
+  applyBrawlerEffects(enemyActiveTraits, enemies);
```
[the remark this answers is not in the document]
Comment (keyy1315) [reply src/lib/simulator/engine/combatLoop.ts]: @chatgpt-codex-connector 지적 감사합니다. `applyBrawlerEffects` 호출을 `applyStargazerEffects` (Huntress mark 선택) 이전 위치로 이동했습니다. Astronaut HP 와 동일 패턴 (이미 P2 회귀 가드로 적용됨) 으로 정렬되어, 적 Brawler 가 활성된 상태에서 마크가 정확한 maxHp 기준으로 선택되도록 보장합니다. Calibration 동일 (23일 45.5% / 24일 76.2%). cf8d36b
